src/guiguts/maintext.py

```diff
@@ -260,6 +265,53 @@ def theme_change(self) -> None:
         self.configure(background=themed_style().lookup("TButton", "background"))
         self.text_color = themed_style().lookup("TButton", "foreground")
 
+    def drag_handler_start(self, evt: tk.Event) -> None:
+        """Handle initial click for drag-select operation."""
+        self.textwidget.focus_set()
+        self.drag_begin_index = self.cursor_index(evt)
+        self.drag_handler(evt)
+
+    def drag_handler(self, evt: tk.Event) -> None:
+        """Select text based on the current drag-select motion event."""
+        _range = self.drag_range(evt)
+        maintext().do_select(_range)
+        self.set_insert_index(evt)
+        maintext().linenumbers.redraw()
+        maintext().peer_linenumbers.redraw()
+
+    def cursor_index(self, evt: tk.Event) -> IndexRowCol:
+        """Return IndexRowCol for the row the cursor is positioned on."""
+        return IndexRowCol(self.textwidget.index(f"@0,{evt.y}"))
+
+    def drag_range(self, evt: tk.Event) -> IndexRange:
+        """Return an IndexRange representing text to select for the
+        drag-selection operation in progress."""
+        sorted_rows = sorted((self.drag_begin_index.row, self.cursor_index(evt).row))
+        # Add 1 to end row to select entire row. Instead of using `lineend`
+        # which would leave a jagged end to our selection, select up to the
+        # 0 index on the following line to get the nice rectangle, selecting
+        # only full lines.
+        return IndexRange(
+            IndexRowCol(sorted_rows[0], 0), IndexRowCol(sorted_rows[1] + 1, 0)
+        )
+
+    def set_insert_index(self, evt: tk.Event) -> None:
+        """Set the insert index so the line number is highlighted properly."""
+        cursor = self.cursor_index(evt)
+        _range = self.drag_range(evt)
+        if self.drag_begin_index.row < cursor.row:
+            # we were dragging downward
+            self.textwidget.mark_set(tk.INSERT, f"{cursor.index()} lineend")
+            self.textwidget.mark_set(TK_ANCHOR_MARK, _range.start.index())
+        elif self.drag_begin_index.row > cursor.row:
+            # we were dragging upward
+            self.textwidget.mark_set(tk.INSERT, _range.start.index())
+            self.textwidget.mark_set(TK_ANCHOR_MARK, f"{_range.end.row}.0")
+        else:
+            # clicked only a single line
+            self.textwidget.mark_set(tk.INSERT, f"{_range.end.index()}-1c")
```

Comment:
Similarly, avoid the insert position being at the end of the single selected line (which isn't a "whole line" select) and putting it at the start of the next line, this should be
```python
self.textwidget.mark_set(tk.INSERT, f"{_range.end.index()}")
```

Reply:
Made this change (but without the string embed)